sllurp/llrp_proto.py

```diff
@@ -1839,11 +1867,16 @@ def encode_ROSpecStartTrigger(par, param_info):
         'PeriodicTriggerValue',
         'GPITriggerValue'
     ],
-    'encode': encode_ROSpecStartTrigger
+    'encode': encode_ROSpecStartTrigger,
+    'decode': basic_auto_param_decode_generator(
+        ubyte_unpack,
+        ubyte_size,
+        'ROSpecStartTriggerType'
+    )
```

Comment:
I think that it is ok for a first version to be merged, but later, we should probably create the right decode function so that we can transform the trigger type to a name with StartTrigger_Type2Name.
Same thing for CurrentState in previous ROSpec and co decoders.

Reply:
check out auto_type2name, let me know what you think.

Reply:
The idea for `auto_type2name` is not bad, but in our case I don't think that it would be a good idea.
Because the `generated_func` is the one that will be called during execution for each parameter encountered, and they were "shaved" to be the minimal possible in order to have the best performances.

In fact, I did think of that kind of thing also for simple cases where we have "conversion" to do, like casting to "int" or just taking a bit flag from a field. But I ruled them out because each if/condition/... could have a huge impact when called thousands of times.

That being said, not all parameters and so encoders/decoders are needed to be called thousands of times. Like the ones for rospec encoding/decoding.
So, I was thinking that maybe we could create a "slow" and "smarter" version of "auto" functions for such cases.
But I was thinking that, so far, the amount of parameters concerned did not worth the trouble, and that still it was a little bit sad to slow down a little bit for performances of "start to inventory" functions just to make it easier for dev.

FYI, as a side note, regarding auto_type2name, if I had to do it, I would probably use a "conversion" mapping dict instead of "string" construction and lookup on global.
For example, some construction that is interesting is to use the type of "arg":
- if simple string (`"ROSpecStartTriggerType"`), do default operation
- if tuple (`("ROSpecStartTriggerType", _conversion_func)`), use tuple additional values to do conversion, operation, select interesting bit of a field, etc....

Reply:
I get what you mean about the performance issue. I did think about it, but thought it's not a big deal.